### ISSUE TITLE
QUIT signals swallowed when authorizations are required

### ISSUE THREAD

**Issue by smaarn** [morpheus65535/bazarr#811]
**Describe the bug**
When authentication is required, trying to shutdown bazarr by sending the `SIGTERM` procedure to the main process doesn't succeed in performing a smooth shutdown on the child process.

**To Reproduce**
1. Enable authorization through forms (`Settings > Security Settings` select `Forms (Login Page)` in `Authentication`) then restart if required
2. Send the `QUIT` (`kill -2`) on the subprocess
3. Check the output of the console

You will see the following stack trace appear:
```

Traceback (most recent call last):
  File "bazarr/main.py", line 2218, in <module>
    shutdown()
  File "bazarr/main.py", line 132, in wrapper
    return func(*a, **ka)
  File "bazarr/main.py", line 207, in shutdown
    authorize()
  File "bazarr/main.py", line 150, in authorize
    aaa.require(fail_redirect=(base_url + 'login'))
  File "/Users/antoine/Projects/bazarr/bazarr/../libs/cork/cork.py", line 186, in require
    cu = self.current_user
  File "/Users/antoine/Projects/bazarr/bazarr/../libs/cork/cork.py", line 345, in current_user
    username = session.get('username', None)
AttributeError: 'NoneType' object has no attribute 'get'
```
And the bazarr process would still be running.
One can note that the web interface part is indeed shut down here.

**Expected behavior**
The bazarr process should exit cleanly.

**Software (please complete the following information):**
 - Bazarr: 0.8.4.1
 - OS: Mac OS X 10.12 (also present on Synology DSM 6.0)

**Additional context**
Basically this issue was spotted while performing tests on "service like" operation on bazarr when running in my DSM. I noticed that when sending the SIGTERM signal to the parent process
1. it would tell me everything was exited 
2. but the child process was still running without being able to access the web interface

**Comment by morpheus65535** [comment]
Merged #810
